dask/dataframe/core.py

```diff
@@ -565,6 +565,89 @@ def std(self, axis=None, ddof=1):
             name = '{0}std(ddof={1})'.format(self._token_prefix, ddof)
             return map_partitions(np.sqrt, None, v, token=name)
 
+    def _cum_agg(self, token, chunk, aggregate, agginit):
+        """ Wrapper for cumulative operation """
+        # cumulate each partitions
+        name1 = '{0}{1}-map'.format(self._token_prefix, token)
+        cumpart = map_partitions(chunk, self.column_info, self, token=name1)
+        # take last element of each cumulated partitions
+        name2 = '{0}{1}-take-last'.format(self._token_prefix, token)
+        cumlast = map_partitions(lambda x: x.iloc[-1],
+                                 self.column_info, cumpart, token=name2)
+
+        name = '{0}{1}'.format(self._token_prefix, token)
+        cname = '{0}{1}-cum-last'.format(self._token_prefix, token)
+
+        # aggregate cumulated partisions and its previous last element
+        dask = {}
+        for i in range(self.npartitions):
+            if i == 0:
+                if isinstance(self, DataFrame):
+                    agginit = pd.Series(agginit, index=self.column_info)
+                dask[(cname, i)] = agginit
+                dask[(name, i)] = (cumpart._name, i)
+            else:
+                # store each cumulative step to graph to reduce computation
+                dask[(cname, i)] = (aggregate, (cname, i - 1),
+                                    (cumlast._name, i - 1))
+                dask[(name, i)] = (aggregate, (cumpart._name, i), (cname, i))
+        return self._constructor(merge(dask, cumpart.dask, cumlast.dask),
+                                 name, self.column_info, self.divisions)
+
+    @wraps(pd.DataFrame.cumsum)
+    def cumsum(self, axis=None):
+        axis = self._validate_axis(axis)
+        if axis == 1:
+            f = lambda x: x.cumsum(axis=1)
+            name = '{0}cumsum(axis=1)'.format(self._token_prefix)
+            return map_partitions(f, self.column_info, self, token=name)
```

Comment:
This can probably be `map_partitions(self._partition_type.cumsum, self.column_info, self, 1, token=name`.  Using the function name will make this slightly easier to debug.